sktime/transformations/series/detrend/_detrend.py

```diff
@@ -84,9 +87,10 @@ class Detrender(BaseTransformer):
         "transform-returns-same-time-index": True,
     }
 
-    def __init__(self, forecaster=None, model="additive"):
+    def __init__(self, forecaster=None, model="additive", out_of_sample=False):
         self.forecaster = forecaster
         self.model = model
+        self.oos = out_of_sample
```

Comment:
the issue is here: you need to write the parameter `out_of_sample` to `self.out_of_sample` - this is a convention from `sklearn`, that all init params need to be written to `self`, with the same name.

Reply:
Got it!